### PR TITLE
feat(demos): use addUrlForHydration for preloading in ssr demo

### PR DESCRIPTION
**Pull request by KingHenne**

### CODE REVIEW

File: packages/demos/src/server-side-rendering/integrator.tsx [hunk 1]
```diff
@@ -19,6 +18,18 @@ function getSerializedStatesFromDom(): string | undefined {
   return (scriptElement && scriptElement.textContent) || undefined;
 }
 
+function getUrlsForHydrationFromDom(): string[] | undefined {
```
Comment (clebert) [inline packages/demos/src/server-side-rendering/integrator.tsx]: Please always return an empty array

Comment (KingHenne) [reply packages/demos/src/server-side-rendering/integrator.tsx]: and then check the length in the guard below? why?

Comment (KingHenne) [reply packages/demos/src/server-side-rendering/integrator.tsx]: the guard is only theoretical anyway, so maybe getting rid of it and returning an empty array here is indeed better

Comment (KingHenne) [reply packages/demos/src/server-side-rendering/integrator.tsx]: fixed (sorry, I accidentally amended)

**Pull request merged**